Outfit list implemented

irvin-solano opened the pull request:
Ticket: https://trello.com/c/kYfLTXix
Related 
- reused Related
- if no product id passed in, render the array outfitList as if it were the related list
-css use to hold a spot on page for outfit list to stop the page from drastically changing when there were some vs none in the lis
- refactored the server call to work for both by making it a helper function (compileLOutfits)

arkteck commented:
It seems like the component is making all its API calls whenever you add/remove to outfit, which can quickly overload the server and seems unnecessary.

irvin-solano commented:
Related
-split up the info into an id array
-an object with the the id as the keys, and the info as the value
(to be able to map through the objects, but still return an array of nodes)
-changed the mapping process for rendering ProductCards to accomodate the use of the outfitId array in conjunction with the outfits object 

-added the new props to all relevant components throught out repo

Product Card
-line 28: created new array, and made mods to it to replace old array when setting the new outfits object
